app/controllers/course/achievement/achievements_controller.rb

```diff
@@ -9,15 +9,11 @@ def index #:nodoc:
   def show #:nodoc:
   end
 
-  def new #:nodoc:
-  end
-
   def create #:nodoc:
     if @achievement.save
-      redirect_to course_achievements_path(current_course),
-                  success: t('.success', title: @achievement.title)
+      render json: { id: @achievement.id }, status: :ok
```

Comment:
In the later commits we use this to redirect to the `show` page for the achievement.